service_configuration_lib/spark_config.py

```diff
@@ -446,6 +448,11 @@ def _get_k8s_spark_env(
     volumes: Optional[List[Mapping[str, str]]],
     paasta_pool: str,
 ) -> Dict[str, str]:
+    # RFC 1123: https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#dns-label-names
+    # technically only paasta instance can be longer than 63 chars. But we apply the normalization regardless.
+    _paasta_cluster = _get_k8s_resource_name_prefix(paasta_cluster)[:DEFAULT_K8S_LABEL_LENGTH]
+    _paasta_service = _get_k8s_resource_name_prefix(paasta_service)[:DEFAULT_K8S_LABEL_LENGTH]
+    _paasta_instance = _get_k8s_resource_name_prefix(paasta_instance)[:DEFAULT_K8S_LABEL_LENGTH]
```

Comment:
might be nice to be consistent with how we shorten things in paasta (I recently fixed this for Tron since I had forgotten about this limit in https://github.com/Yelp/paasta/pull/3208/files) - https://github.com/Yelp/paasta/blob/f65ecfa6d6b5d3074951c640e4bf2b8bc16b826d/paasta_tools/kubernetes_tools.py#L409:5

and while we're looking at labels in this PR: it's probably worth going through y/paasta-workload-contract and making sure that the other labels specified there are being set (as well as some of the other k8s settings)